spec/models/activity_spec.rb

```diff
@@ -653,4 +653,15 @@
       expect(project.actual_total_for_report_financial_quarter(report: report)).to eq(100.20)
     end
   end
+
+  describe "#forecasted_total_for_report_financial_quarter" do
+    it "returns the total of all the activity's planned disbursements scoped to a report's financial quarter" do
```

Comment:
I really wish there was some way to express/document the behaviour of these totalling methods, I just can't come up with it! I got to this yesterday, but I think it makes it look even more complicated:

````ruby
    it "totals the activity forecasts for the financial quarter of the report" do
      quarter_one_date = Date.parse("1 April 2019")
      quarter_two_date = Date.parse("1 July 2019")

      project = create(:project_activity)
      create(:planned_disbursement, parent_activity: project, value: 1000.00, period_start_date: quarter_one_date)
      create(:planned_disbursement, parent_activity: project, value: 1000.00, period_start_date: quarter_one_date)

      create(:planned_disbursement, parent_activity: project, value: 2000.00, period_start_date: quarter_two_date)
      create(:planned_disbursement, parent_activity: project, value: 2000.00, period_start_date: quarter_two_date)

      report = create(:report, fund: project.associated_fund)
      expect(project.forecasted_total_for_report_financial_quarter(report: report)).to eq(0)

      travel_to(quarter_one_date) do
        report = create(:report, fund: project.associated_fund)
        expect(project.forecasted_total_for_report_financial_quarter(report: report)).to eq(2000.00)
      end

      travel_to(quarter_two_date) do
        report = create(:report, fund: project.associated_fund)
        expect(project.forecasted_total_for_report_financial_quarter(report: report)).to eq(4000.00)
      end
    end
````

I feel like these are important so I expect more tests, but maybe I am given them too much credit?

Either way, we need to move this work forward and I hope once folks understand what we are aiming for your tests make sense - let me know your thoughts.

Reply:
I've included your tests almost verbatim as they make perfect sense to me, much more thorough than the single one I had.